lib/12T/magic/ext4mag.py

```diff
@@ -0,0 +1,195 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License. 
+
+"""Script Description
+	This script performs extraction from the magic layout files. This script extracts the 
+	.sim, .spice, .gds, .ext, and .cif files from the magic files. Please run this script 
+	inside of the magic files directory to extract.
+ 
+	Utilize the following terminal format(s):
+	
+	The following format extracts all files from all .mag files without moving them to separate directories:
+	python3 ext4mag.py 
+	
+	The following format extracts all files and moves them to separate directories:
+	python3 ext4mag.py -move
+
+	The following format extracts all files, moves them to separate directories, and extracts
+	ngspice files:
+	python3 ext4mag.py -move -ng 
+
+	The following format extracts all files, moves them to separate directories, and extracts
+	ngspice files, and adds library inclusion commands to the spice files to include the appropriate models:
+	python3 ext4mag.py -move -ng -ap
+	
+	The following format extracts all files for a single .mag file. Note that all above file
+	combinations can be applied to the following command: 
+	python3 ext4mag.py file_name.mag
+
+	By default, the Makefile included in this directory will run the following format:
+	python3 ext4mag.py -move -ap -ng
+
+	Note: This script is intended to be used inside of the magic cell library directory. If
+	the library directory is moved, please change the directory paths below appropriately to ensure a
+	successful script execution.
+"""
+
+import subprocess
+import traceback
+import shutil
+import sys
+import os
+
+class Ext4Mag():
+	def __init__(self):
+		self.files = os.listdir()
+		self.move_flag = False
+		self.add_params = False
+		self.exng = False 
+		self.enter = True
+		self.cwd = os.getcwd()
+
+		os.chdir('../../char/techfiles/')
+
+		self.pdk_spice = os.getcwd()
+		os.chdir(self.cwd)
+
+		if (len(sys.argv) > 1):
+			self.cmdl_parse(sys.argv)
+
+	def cmdl_parse(self, args):
+		for i in range(len(args)):
+			if '.mag' in args[i]:
+				if self.enter:
+					self.enter = False
+					self.files = []
+				self.files.append(args[i])
+			if '-move' in args[i]:
+				self.move_flag = True
+			if '-ap' in args[i]:
+				self.add_params = True
+			if '-ng' in args[i]:
+				self.exng = True
+				
+	
+	def move_files(self):
+		names = ['.sim', '.spice', '.gds', '.ext', '.cif']
+		files = os.listdir()
+
+		for f in files:
+			for s in names:
+				if s in f and '.magicrc' not in f:
+					if not os.path.exists(s.split(".")[1]):
+						os.mkdir(s.split(".")[1])
+					shutil.move(f'./{f}', f'./{s.split(".")[1]}/{f}')
+	
+	def run_magic(self):
+		for f in self.files:
+			if '.mag' in f and '.magicrc' not in f:
+				print(f'Current magic file: {f}')
```

Comment:
did you consider using https://docs.python.org/3/library/logging.html to be able to level the verbosity of the logs?

Reply:
deleted - not needed.